src/v/archival/service.cc

```diff
@@ -234,48 +234,62 @@ ss::future<> scheduler_service_impl::add_ntp_archiver(
     if (_gate.is_closed()) {
         return ss::now();
     }
-    return archiver->download_manifest().then(
-      [this, archiver](cloud_storage::download_result result) {
-          auto ntp = archiver->get_ntp();
-          switch (result) {
-          case cloud_storage::download_result::success:
-              vlog(
-                _rtclog.info,
-                "Found manifest for partition {}",
-                archiver->get_ntp());
-              _probe.start_archiving_ntp();
-
-              _archivers.emplace(archiver->get_ntp(), archiver);
-              archiver->run_upload_loop();
-
-              return ss::now();
-          case cloud_storage::download_result::notfound:
-              vlog(
-                _rtclog.info,
-                "Start archiving new partition {}",
-                archiver->get_ntp());
-              // Start topic manifest upload
-              // asynchronously
-              if (ntp.tp.partition == 0) {
-                  // Upload manifest once per topic. GCS has strict
-                  // limits for single object updates.
-                  (void)upload_topic_manifest(
-                    model::topic_namespace(ntp.ns, ntp.tp.topic),
-                    archiver->get_revision_id());
-              }
-              _probe.start_archiving_ntp();
-
-              _archivers.emplace(archiver->get_ntp(), archiver);
-              archiver->run_upload_loop();
+    return archiver->download_manifest().then([this, archiver](
+                                                cloud_storage::download_result
+                                                  result) {
+        auto ntp = archiver->get_ntp();
+        auto part = _partition_manager.local().get(ntp);
+        switch (result) {
+        case cloud_storage::download_result::success:
+            vlog(
+              _rtclog.info,
+              "Found manifest for partition {}",
+              archiver->get_ntp());
+
+            if (part->get_ntp_config().is_read_replica_mode_enabled()) {
+                archiver->run_sync_manifest_loop();
+            } else {
+                _probe.start_archiving_ntp();
+                archiver->run_upload_loop();
+            }
+            _archivers.emplace(archiver->get_ntp(), archiver);
+
+            return ss::now();
+        case cloud_storage::download_result::notfound:
+            if (part->get_ntp_config().is_read_replica_mode_enabled()) {
+                vlog(
+                  _rtclog.info,
+                  "Couldn't download manifest for partition {} in read replica",
+                  archiver->get_ntp());
+                // todo: throw here?
```

Comment:
is this todo resolved?

Reply:
@Lazin do you think it's worth throwing an exception here? cause if the partition manifest is not found that means that the data is not in s3 or the configuration is wrong.

Reply:
I don't think so. This could mean that the manifest wasn't uploaded **yet**.  For instance, some partition may have a single segment that didn't roll yet but other partitions of the topic might already do this. So if you will create a read replica without pushing enough data to the topic you will trigger this condition. So I think that it's OK to just log something here.

Reply:
We should test this. 
1. Create an original topic
2. Create read replica
3. Push data to original topic
4. Read from read replica

Reply:
tested it manually, works as expected